bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -189,6 +189,89 @@
 
 	</thing-type>
 
+	<thing-type id="intrusion-detection-system">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="shc"/>
+		</supported-bridge-type-refs>
+
+		<label>Intrusion Detection System</label>
+		<description>Allows to retrieve and control the state of the intrusion detection alarm system.</description>
+
+		<channels>
+			<channel id="system-availability" typeId="system-availability"/>
+			<channel id="arming-state" typeId="arming-state"/>
+			<channel id="alarm-state" typeId="alarm-state"/>
+			<channel id="active-configuration-profile" typeId="active-configuration-profile"/>
+			<channel id="arm-action" typeId="arm-action"/>
+			<channel id="disarm-action" typeId="disarm-action"/>
+			<channel id="mute-action" typeId="mute-action"/>
+		</channels>
+
+	</thing-type>
+
+	<channel-type id="system-availability">
+		<item-type>Switch</item-type>
+		<label>System Availability</label>
+		<description>Indicates whether the intrusion detection system is available.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="arming-state">
+		<item-type>String</item-type>
+		<label>Arming State</label>
+		<description>The arming state of the intrusion detection system. Possible values are SYSTEM_ARMING, SYSTEM_ARMED and
+			SYSTEM_DISARMED. This channel is read-only. Use the arm-action and disarm-action channels to arm and disarm the
+			system.</description>
+		<state readOnly="true">
+			<options>
+				<option value="SYSTEM_ARMING">System is currently arming</option>
+				<option value="SYSTEM_ARMED">System is armed</option>
+				<option value="SYSTEM_DISARMED">System is disarmed</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="alarm-state">
+		<item-type>String</item-type>
+		<label>Alarm State</label>
+		<description>The alarm state of the intrusion detection system. Possible values are ALARM_OFF, PRE_ALARM, ALARM_ON,
+			ALARM_MUTED and UNKNOWN.</description>
+		<state readOnly="true">
+			<options>
+				<option value="ALARM_OFF">No alarm</option>
+				<option value="PRE_ALARM">Alarm is about to go off</option>
+				<option value="ALARM_ON">Alarm was triggered</option>
+				<option value="ALARM_MUTED">Alarm is muted</option>
+				<option value="UNKNOWN">Alarm status is unknown</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="active-configuration-profile">
+		<item-type>String</item-type>
+		<label>Active Configuration Profile</label>
+		<description>The name of the active configuration profile used for the intrusion detection system.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="arm-action">
+		<item-type>String</item-type>
+		<label>Arm Action</label>
+		<description>Arms the intrusion detection system using the given profile ID.</description>
+	</channel-type>
+
```

Comment:
I was wondering if some channels could be combined to one, for example the arm action and the disarm action. However, arming requires a string command (the profile to use) and disarming would then require handling for a special string like "OFF". Do you agree that using two separate channels is better in this case?

Reply:
IMHO it's fine the way it is.

Reply:
I'm fine sticking close to the SHC API, because we avoid own logic in the Bosch SHC binding . Either the API is providing already the right level of abstrasction or the users can use e.g. openhab rules if needed.